flink-runtime-web/web-dashboard/src/app/share/customize/logs-bundler/logs-bundler.component.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnInit} from '@angular/core';
+import {flatMap, takeUntil} from "rxjs/operators";
+import { Observable, Subject} from "rxjs";
+import { StatusService} from "services";
+import {LogsBundlerService} from "../../../services/logs-bundler.service";
+import {BASE_URL} from "config";
+import {LogsBundlerStatus} from "../../../interfaces/logs-bundler";
+
+@Component({
+    selector: 'flink-logs-bundler',
+    templateUrl: './logs-bundler.component.html',
+    styleUrls: [],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LogsBundlerComponent implements OnInit{
+    destroy$ = new Subject();
+    @Input() statusObservable: Observable<LogsBundlerStatus>;
+    hideSpinner: boolean = true;
+    message: string = ""
+    hideDownloadButton: boolean = true;
+
+    constructor(private logBundlerService: LogsBundlerService,
+                private statusService: StatusService,
+                private cdr: ChangeDetectorRef) {
+    }
+
+    ngOnInit() {
+        this.statusObservable = this.statusService.refresh$
```

Comment:
There's actually no reason to even have `statusObservable` as a member field at all. You can just use a local variable.

```
const status$ = this.statusService.…;
status$.subscribe(…);
```

Since nothing much happens here with that variable you can even just combine it altogether:

```
this.statusService.refresh$.pipe(
  flatMap(() => this.logBundlerService.getStatus()),
  takeUntil(this.destroy$),
).subscribe(status => {
  // …
});
```

Reply:
There's also a whole thing about whether this should use `flatMap`, `concatMap` or `switchMap`, as all of these have slightly different semantics. With `mergeMap`, if requests are too slow, we'll end up running them in parallel and stitching them together in whichever order they finish, including running out of order. With `switchMap` we avoid this, but it could insted lead to it never emitting at all. As this is an unlikely scenario it doesn't matter, but just pointing out the considerations that typically would go into the choice of operator.